api/src/main/java/com/graphhopper/json/Clause.java

```diff
@@ -18,65 +18,61 @@
 package com.graphhopper.json;
 
 public class Clause {
-    // either if or else-if with thenValue
-    private String ifClause;
-    private String elseIfClause;
-    private Double thenValue;
-    // or just elseValue
-    private Double elseValue;
+    private final Cond condition;
+    private final String expression;
```

Comment:
Isn't `expression` the 'condition' here and condition is a ... hmmm ... 'clause' :-D I mean `if/else` are 'clauses' (they aren't really 'conditions') and something like `road_class == MOTORWAY` is a 'condition'. Ok the latter is also an 'expression', but 'condition' is more specific, it always evaluates to a boolean.

Reply:
Yeah, it was a bit confusing. But the current state is better I think?